R/Registry.R

```diff
@@ -70,6 +70,73 @@ Registry <- R6::R6Class( # nolint object_name_linter
       private$.record_class$new(data = data)
     },
     #' @description
+    #' Get a data frame summarising records in the registry
+    #'
+    #' @param limit Maximum number of records to return
+    #' @param verbose Boolean, whether to print progress messages
+    #'
+    #' @return A data.frame containing the available records
+    df = function(limit = 100, verbose = FALSE) {
+      # The API is limited to 200 records at a time so we need multiple requests
+      n_requests <- ceiling(limit / 200)
+      if ((verbose && n_requests > 1) || n_requests >= 10) {
+        caution <- ifelse(n_requests >= 10, "CAUTION: ", "")
+        cli::cli_alert_warning(
+          "{caution}Retrieving {limit} records will require up to {n_requests} API requests"
+        )
+      }
+
+      data_list <- list()
+      attr(data_list, "finished") <- FALSE
+      data_list <- purrr::reduce(
+        cli::cli_progress_along(seq_len(n_requests), name = "Sending requests"),
+        \(.data_list, .n) {
+          # Hacky way of avoiding unneeded requests until there is an easy way
+          # to get the total number of records
+          if (isTRUE(attr(.data_list, "finished"))) {
+            return(.data_list)
+          }
```

Comment:
If there is a way to get the total number of records we can use a regular `purrr::map` instead